addons/account_asset/models/account_asset.py

```diff
@@ -103,6 +112,21 @@ class AccountAssetAsset(models.Model):
         help="It is the amount you plan to have that you cannot depreciate.")
     invoice_id = fields.Many2one('account.invoice', string='Invoice', states={'draft': [('readonly', False)]}, copy=False)
     type = fields.Selection(related="category_id.type", string='Type', required=True)
+    date_first_depreciation = fields.Selection([
+        ('first_day_month', 'Purchase Date'),
```

Comment:
why is the key 'first_day_month' if it means 'purchase_date' ?

here is how it is supposed to work: 
```python
      help='The method to compute the date of the first depreciation.'
            'If set as Purchase Date, it will take the accounting date set on the supplier bill.'
            'In case of Last Day of Period, it will either take the last day of the month or fiscalyear accordingly'
            'to the number of period in one year set.'
            'Manual will allow you to set a custom date for the first depreciation.')
```